test/mpi/runtests.in

```diff
@@ -315,8 +316,11 @@ if ($listfiles eq "") {
         &ProcessImplicitList;
     }
 }
-elsif (-d $listfiles) { 
-    print STDERR "Testing by directories not yet supported\n";
+elsif ($testdirs ne "") {
+    my @all_testdirs = split /,\s*/, $testdirs;
+    foreach my $_d (@all_testdirs) {
+        &ProcessDir( $_d, $listfiles );
```

Comment:
This only works with top-level dirs. What about e.g. `f77/coll`?

Reply:
It depends on `$listfiles`. If configured to use `testlist`, then it would pickup sub-directories listed in `f77/testlist` including `f77/coll`.

Reply:
I removed the limitation to only allow top-level directories in ProcessDir. In fact, the warning was wrong, subdirectories were perfectly fine as it was. I did make it an error if `chdir` failed, though.